src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -52,8 +45,16 @@ private static void HandleStartArray(JsonSerializerOptions options, ref ReadStac
 
             state.Current.CollectionPropertyInitialized = true;
 
-            // We should not be processing custom converters here (converters are of ClassType.Value).
-            Debug.Assert(state.Current.JsonClassInfo.ClassType != ClassType.Value);
+            // The current JsonPropertyInfo will be null if the current type is not one of
+            // ClassType.Value | ClassType.Enumerable | ClassType.Dictionary.
+            // We should not see ClassType.Value here because we handle it on a different code
+            // path invoked in the main read loop.
+            // Only ClassType.Enumerable is valid here since we just saw a StartArray token.
+            if (state.Current.JsonPropertyInfo == null ||
+                (ClassType.Enumerable & state.Current.JsonPropertyInfo.ClassType) == 0)
```

Comment:
Can this be changed to `==`  It looks like ClassType is only an enum and not Flags.

Reply:
I am not sure. @steveharter - can it?

Even though ClassType isn't a flags enum, it is treated like one for "fast" checks:
https://github.com/dotnet/corefx/blob/8c92fa400f7a96d08c0d35cf14d91446e58d3512/src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs#L10-L14

Reply:
It can. My understanding is `==` is faster when we are comparing against just one `ClassType` value.